cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/ByteBufferUtils.java

```diff
@@ -93,25 +86,27 @@ public static byte[] getArray(ByteBuffer buffer)
         return bytes;
     }
 
-    public static String stringThrowRuntime(ByteBuffer buffer)
+    /**
+     * Decode ByteBuffer into String using provided CharsetDecoder.
+     *
+     * @param buffer          byte buffer
+     * @param decoderSupplier let the user provide their own CharsetDecoder provider
+     *                        e.g. using io.netty.util.concurrent.FastThreadLocal over java.lang.ThreadLocal
+     * @return decoded string
+     * @throws CharacterCodingException charset decoding exception
+     */
+    public static String string(ByteBuffer buffer, Supplier<CharsetDecoder> decoderSupplier) throws CharacterCodingException
     {
-        try
-        {
-            return ByteBufferUtils.string(buffer);
-        }
-        catch (CharacterCodingException exception)
+        if (buffer.remaining() <= 0)
         {
-            throw new RuntimeException(exception);
+            return EMPTY_STRING;
         }
+        return decoderSupplier.get().decode(buffer.duplicate()).toString();
```

Comment:
Nit: I personally would use a conditional operator here for improved readability:
```java
return buffer.remaining() > 0 ? decoderSupplier.get().decode(buffer.duplicate()).toString() : EMPTY_STRING;
```